src/OpenTK.Windowing.GraphicsLibraryFramework/JoystickState.cs

```diff
@@ -0,0 +1,235 @@
+//
+// JoystickState.cs
+//
+// Copyright (C) OpenTK
+//
+// This software may be modified and distributed under the terms
+// of the MIT license. See the LICENSE file for details.
+
+using System;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Text;
+using OpenTK.Windowing.Common.Input;
+
+[assembly: InternalsVisibleTo("OpenTK.Windowing.Desktop")]
+
+namespace OpenTK.Windowing.GraphicsLibraryFramework
+{
+    /// <inheritdoc />
+    public sealed class JoystickState : IJoystickState
```

Comment:
If we're not exposing this directly, how about an internal class?

Reply:
Yes, OK. Since I have InternalsVisibleTo OpenTK.Windowing.Desktop, that will work.